cpp/dolfin/fem/utils.cpp

```diff
@@ -512,6 +512,17 @@ fem::get_constants_from_ufc_form(const ufc_form& ufc_form)
   return constants;
 }
 //-----------------------------------------------------------------------------
+std::shared_ptr<fem::Form> fem::create_form(
+    ufc_form* (*fptr)(void),
```

Comment:
What is `fptr`?

Reply:
`/// @param[in] fptr pointer to a function returning a pointer to ufc_form.`

Reply:
It's a bit cryptic.

I could also mention that the function it points to takes no arguments.

Reply:
That should be clear from the signature `(void)` - but I admit I don't really like C-style function pointer signatures, as they are... cryptic.

Reply:
Use std::function<ufc_form*(void)> as argument type and rely on implicit conversion when a C-style function is provided (no difference for caller)?